analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/IsPattern.cs

```diff
@@ -100,19 +100,28 @@ private static SymbolicConstraint ConstraintFromDeclarationPattern(IDeclarationP
         }
     }
 
-    private static BoolConstraint BoolContraintFromConstant(ProgramState state, IIsPatternOperationWrapper isPattern)
+    private static BoolConstraint BoolConstraintFromConstant(ProgramState state, IIsPatternOperationWrapper isPattern)
     {
         if (state[isPattern.Value] is { } value
             && isPattern.Pattern.WrappedOperation.Kind == OperationKindEx.ConstantPattern
-            && IConstantPatternOperationWrapper.FromOperation(isPattern.Pattern.WrappedOperation).Value.ConstantValue.Value is bool boolConstant)
+            && IConstantPatternOperationWrapper.FromOperation(isPattern.Pattern.WrappedOperation).Value is var constantPattern
+            && constantPattern.ConstantValue.Value is { } constant)
         {
-            if (value.Constraint<BoolConstraint>() is { } valueConstraint)
+            if (constant is bool boolConstant)
             {
-                return BoolConstraint.From(valueConstraint == BoolConstraint.From(boolConstant));
+                if (value.Constraint<BoolConstraint>() is { } valueBool)
+                {
+                    return BoolConstraint.From(valueBool == BoolConstraint.From(boolConstant));
+                }
+                else if (value.HasConstraint(ObjectConstraint.Null))
+                {
+                    return BoolConstraint.False;
+                }
             }
-            else if (value.HasConstraint(ObjectConstraint.Null))
+            else if (state.Constraint<NumberConstraint>(constantPattern) is { } constantNumber
+                && value.Constraint<NumberConstraint>() is { } valueNumber)
```

Comment:
please align the conditions in lines `105-108` to be 4 spaces less like here.

Reply:
Alignment looks good to me. `105-108` have 4 spaces less, because it's the outer `if`.